sdk/workloads/Azure.ResourceManager.Workloads/tests/Tests/SviCRUDTests.cs

```diff
@@ -0,0 +1,235 @@
+        }
+
+        /// <summary>
+        /// Gets an object of SAP Workload Resource for Single Server that should be used for PUT call.
+        /// </summary>
+        /// <param name="infraRgName"> Contains the infra RG
+        /// used as app resource group. </param>
+        /// <returns>PHP Workload Resource.</returns>
+        private SapVirtualInstanceData GetSingleServerPayloadToPut(string infraRgName, bool isInstall)
+        {
+            string sshPrivateKey = File.ReadAllText(Path.Combine(
+                Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SshKeyPrivate.txt"));
+
+            string sshPublicKey = File.ReadAllText(Path.Combine(
+                Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SshKeyPublic.txt"));
+
+            if (isInstall)
+            {
+                var testSapWorkloadJson = File.ReadAllText(Path.Combine(
+                    Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SingleServerInstall.json"));
+
+                dynamic DynamicObject = JsonConvert.DeserializeObject<dynamic>(testSapWorkloadJson);
+
+                DynamicObject.properties.configuration.infrastructureConfiguration.appResourceGroup =
+                    infraRgName;
+                DynamicObject.properties.configuration.infrastructureConfiguration.
+                    virtualMachineConfiguration.osProfile.osConfiguration.ssh.publicKeys[0].keyData =
+                        sshPublicKey;
+                DynamicObject.properties.configuration.softwareConfiguration.sshPrivateKey =
+                    sshPrivateKey;
+
+                var sapPayloadJson = JsonDocument.Parse(JsonConvert.SerializeObject(DynamicObject)).RootElement;
+
+                SapVirtualInstanceData testSapWorkload = SapVirtualInstanceData.DeserializeSapVirtualInstanceData(
+                    sapPayloadJson);
+
+                return testSapWorkload;
+            }
+            else
+            {
+                var testSapWorkloadJson = File.ReadAllText(Path.Combine(
+                    Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SingleServerCreate.json"));
+
+                dynamic DynamicObject = JsonConvert.DeserializeObject<dynamic>(testSapWorkloadJson);
+
+                DynamicObject.properties.configuration.infrastructureConfiguration.appResourceGroup =
+                    infraRgName;
+                DynamicObject.properties.configuration.infrastructureConfiguration.
+                    virtualMachineConfiguration.osProfile.osConfiguration.ssh.publicKeys[0].keyData =
+                        sshPublicKey;
+
+                var sapPayloadJson = JsonDocument.Parse(JsonConvert.SerializeObject(DynamicObject)).RootElement;
+
+                SapVirtualInstanceData testSapWorkload = SapVirtualInstanceData.DeserializeSapVirtualInstanceData(
+                    sapPayloadJson);
+
+                return testSapWorkload;
+            }
+        }
```

Comment:
Something like below is better with less repetition of code

```suggestion
        private SapVirtualInstanceData GetSingleServerPayloadToPut(string infraRgName, bool isInstall)
        {
            string sshPublicKey = File.ReadAllText(Path.Combine(
                Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SshKeyPublic.txt"));
            string fileName = isInstall? "SingleServerInstall.json":"SingleServerCreate.json";

            var testSapWorkloadJson = File.ReadAllText(Path.Combine(
                Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), fileName));

            dynamic DynamicObject = JsonConvert.DeserializeObject<dynamic>(testSapWorkloadJson);

            DynamicObject.properties.configuration.infrastructureConfiguration.appResourceGroup =
                infraRgName;
            DynamicObject.properties.configuration.infrastructureConfiguration.
                virtualMachineConfiguration.osProfile.osConfiguration.ssh.publicKeys[0].keyData =
                    sshPublicKey;

            if(isInstall){
                string sshPrivateKey = File.ReadAllText(Path.Combine(
                    Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SshKeyPrivate.txt"));
                DynamicObject.properties.configuration.softwareConfiguration.sshPrivateKey =
                    sshPrivateKey;
            }

            var sapPayloadJson = JsonDocument.Parse(JsonConvert.SerializeObject(DynamicObject)).RootElement;

            return SapVirtualInstanceData.DeserializeSapVirtualInstanceData(sapPayloadJson);
        }
```

Reply:
Tested this code, works fine. Updated with this, thanks for the suggestion.